src/classes/Database.php

```diff
@@ -65,33 +64,76 @@ public function connect(): \PDO {
         }
     }
 
-    private function createTables(): void {
-        Logger::debug("Creating database tables");
 
-        $sql = "CREATE TABLE IF NOT EXISTS users (
-            id INTEGER PRIMARY KEY AUTOINCREMENT,
-            username TEXT NOT NULL UNIQUE,
-            password TEXT NOT NULL,
-            created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
-        )";
+    public function getMigration(int $version): ?\Cronbeat\Migration {
+        $migrationFile = APP_DIR . '/migrations/' . sprintf('%04d', $version) . '.php';
 
-        if ($this->pdo === null) {
-            $this->connect();
+        if (!file_exists($migrationFile)) {
+            Logger::error("Migration file not found", ['version' => $version, 'file' => $migrationFile]);
+            return null;
         }
 
+        require_once $migrationFile;
+
+        $className = '\\Cronbeat\\Migrations\\Migration' . sprintf('%04d', $version);
+
+        if (!class_exists($className)) {
+            Logger::error("Migration class not found", ['version' => $version, 'class' => $className]);
+            return null;
+        }
         try {
-            if ($this->pdo === null) {
-                throw new \RuntimeException("Failed to connect to database");
+            $migration = new $className();
+
+            if (!$migration instanceof \Cronbeat\Migration) {
+                Logger::error("Invalid migration class", ['version' => $version, 'class' => $className]);
+                return null;
             }
 
-            $this->pdo->exec($sql);
-            Logger::info("Database tables created successfully");
-        } catch (\PDOException $e) {
-            Logger::error("Failed to create database tables", ['error' => $e->getMessage()]);
-            throw $e;
+            return $migration;
+        } catch (\Exception $e) {
+            Logger::error("Error creating migration instance", [
+                'version' => $version,
+                'class' => $className,
+                'error' => $e->getMessage()
+            ]);
+            return null;
         }
     }
 
+    /** @return array<int, \Cronbeat\Migration> */
+    public function getAllMigrations(): array {
```

Comment:
@jetbrains-junie move this to a migration helper class, rename to "loadAllMigrations"